src/Services/ReactionLimiterService.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using Discord;
+using Discord.WebSocket;
+
+namespace BrackeysBot.Services
+{
+    public class ReactionLimiterService : BrackeysBotService, IInitializeableService
+    {
+        private readonly DiscordSocketClient _discord;
+        private readonly DataService _dataService;
+
+        public ReactionLimiterService(
+            DiscordSocketClient discord,
+            DataService dataService)
+        {
+            _discord = discord;
+            _dataService = dataService;
+        }
+        
+        public void Initialize()
+        {
+            _discord.ReactionAdded += CheckReactionAsync;
+        }
+
+        public async Task CheckReactionAsync(Cacheable<IUserMessage, ulong> cacheable, ISocketMessageChannel channel, SocketReaction reaction) 
+        {
+            Dictionary<ulong, List<string>> restricted = _dataService.Configuration.EmoteRestrictions;
+
+            if (restricted != null) 
+            {
+                List<string> limitedToEmotes = restricted.GetValueOrDefault(channel.Id);
+
+                if (limitedToEmotes.Count > 0 && !limitedToEmotes.Contains(reaction.Emote.StringVal())) 
+                    await reaction.Message.Value.RemoveReactionAsync(reaction.Emote, reaction.User.Value);

```

Comment:
Does this always work? Are there complications with rate limitations?

Reply:
I don't see us using this in the announcement channel. The target channel has no rate limitation issues I'd imagine.

Worst case we get rate limited and it is deleted belated.